core/src/main/java/com/aem/demo/core/services/impl/RssFeedServiceImpl.java

```diff
@@ -169,4 +175,25 @@ private Set<String> getSavedNodeNames(String rootPath, PageManager pageManager)
         return names;
     }
 
+    private String[] makeTags(ResourceResolver resourceResolver, NewsCard card) {
+        String[] pathArr = new String[3];
+        List<String> titles = splitTitle(card.getTopic());
+        TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+        try {
+            for (int i = 0; i < titles.size(); i++) {
+                String title = titles.get(i);
+                Tag tag = tagManager.createTag(title, title, card.getTopic(), true);
+                pathArr[i] = tag.getPath();
+            }
+        } catch (InvalidTagFormatException e) {
+            log.error("Exception occurred while creating tags " + e.getMessage());
+        }
+        return pathArr;
+    }
+
+    private List<String> splitTitle(String title) {
+        String regex = "([,\\s?.'’@]+)|(the)|(for)|(with)|(from)|(by)|(gets)";
```

Comment:
You don't need to enclose every "part" of regexp in `(  )` This way, you produce "capturing groups" (the groups that can be referred to further on) but you don't need them

Reply:
And surely you need a wider list of "service words". Article "the" is here but where is "a"? "For" is here but where is "by", "in", "on", "at", etc.
Please remember you were suggested to throw away all the words with the number of letters <= 3